homeassistant/components/shelly/__init__.py

```diff
@@ -43,19 +47,19 @@
 )
 from .utils import get_coap_context, get_device_name, get_device_sleep_period
 
-PLATFORMS = ["binary_sensor", "cover", "light", "sensor", "switch"]
-SLEEPING_PLATFORMS = ["binary_sensor", "sensor"]
-_LOGGER = logging.getLogger(__name__)
+PLATFORMS: Final[list[str]] = ["binary_sensor", "cover", "light", "sensor", "switch"]
```

Comment:
I'm kinda against marking everything final for no apparent reason.

Additionally, there is no need to specify a type in that case, as the type can be extract from the value (as it is final at this point anyways).

Reply:
You're right that there's no need to specify type in this case. I checked with `reveal_type` calls that mypy can infer proper type in such cases.

But marking constants as `Final` makes sense to me as an extra type safety check to protect constants from re-assignment. Otherwise there're no automatic checks for this and only manual review can catch such things. Note that otherwise constants can be modified from different package (e.g. another component) which may be reviewed  only by component authors.

Although this is rare but I've seen example when one package modified constants inside `ssl` package causing issues for different package which also uses ssl.

IMO, having such annotations doesn't hurt and the only downside is some extra noise.

Reply:
So what is the correct approach to the type of constants? I don't know if I should change it or not.

Reply:
I am good to go with everything in this PR, we only need to make the decision of the constants

Reply:
Well, the Final... dunno, I think they are spammy if applied everywhere, but it is correct.

The additional typing when using Final is not needed, as that is inferred from the value.